tools/build-tools/src/typeValidator/testGeneration.ts

```diff
@@ -87,14 +87,25 @@ interface TestCaseTypeData extends TypeData{
 function buildTestCase(getAsType:TestCaseTypeData, useType:TestCaseTypeData, isCompatible: boolean){
     const getSig =`get_${getAsType.prefix}_${getFullTypeName(getAsType).replace(".","_")}`;
     const useSig =`use_${useType.prefix}_${getFullTypeName(useType).replace(".","_")}`;
+    const getIsRemoved = getAsType.kind.startsWith("Removed");
+    const useIsRemoved = useType.kind.startsWith("Removed");
     const testString: string[] =[];
-    testString.push(`declare function ${getSig}():\n    ${toTypeString(getAsType.prefix, getAsType)};`);
-    testString.push(`declare function ${useSig}(\n    use: ${toTypeString(useType.prefix, useType)});`);
+
+    testString.push(`declare function ${getSig}():`);
+    if (!isCompatible && getIsRemoved) {
+        testString.push(`    // @ts-expect-error compatibility expected to be broken`);
+    }
+    testString.push(`    ${toTypeString(getAsType.prefix, getAsType)};`);
+    testString.push(`declare function ${useSig}(`);
+    if (!isCompatible && useIsRemoved) {
+        testString.push(`    // @ts-expect-error compatibility expected to be broken`);
+    }
+    testString.push(`    use: ${toTypeString(useType.prefix, useType)});`);
     testString.push(`${useSig}(`);
-    if(!isCompatible){
+    if(!isCompatible && !getIsRemoved && !useIsRemoved) {
```

Comment:
i don't think we care about removed in this case. if it is not compatible, we should always comment it out

Reply:
I wasn't sure about this either, but I had noticed that the compiler would complain about the ts directive being unused. It seems like in this scenario `getSig()` does not produce any compilation errors (which does seem strange).

Reply:
or does that create a problem due to the next this line not having an error, so the comment creates an error? if that's the case, can you add a commend to the code about what it is necessary

Reply:
missed your comment. i think i get it. since we say we know the declaration has an error, typescript probably just treats it as an `any` that make the below code valid, as typescript doesn't really check anything with an any.